src/hooks/useLocalStorage.ts

```diff
@@ -0,0 +1,14 @@
+import { useEffect, useState } from 'react';
+
+export const useLocalStorage = (key: string, initialValue: string) => {
+  const [storedValue, setStoredValue] = useState(() => {
+    const value = localStorage.getItem(key);
+    return value || initialValue;
+  });
+
+  useEffect(() => {
+    localStorage.setItem(key, storedValue.trim());
```

Comment:
It's a bit strange favor, as it leads to the next situation:
```
set(key, value);
console.log(value === get(key)); // false
```

But if you really want it you don't need an effect, just wrap `setStoredValue` into another function that does whatever you want over the input value and calls the state dispatch.